src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -981,8 +981,10 @@ public function __construct()
         if ($this->container->isCompiled()) {
             if (Container::class !== $baseClassWithNamespace) {
                 $r = $this->container->getReflectionClass($baseClassWithNamespace, false);
-
-                if (null !== $r && (null !== $constructor = $r->getConstructor()) && 0 === $constructor->getNumberOfRequiredParameters()) {
+                if (null !== $r
+                    && $r->getMethod('__construct')->class === $baseClassWithNamespace
```

Comment:
we use `$constructor = $r->getConstructor()` to get the constructor (and your code here breaks in case there is no constructor). Add this logic after the retrieval of `$constructor`.

And I would check instead `Container::class !== $contructor->getDeclaringClass()`, to exclude only the Container constructor.

Reply:
you mean `$constructor->getDeclaringClass() === Container::class` ?